app/controllers/donors_controller.rb

```diff
@@ -11,6 +11,7 @@ def index
               else
                 Donor.all
               end
+    @donors = @donors.includes(:donations).where("donations.cause_id = ?", params[:cause_id]).references(:donations) if params[:cause_id].present?
```

Comment:
Well done! 😀 

I think this controller action is getting complex enough that it warrants being wrapped in its own Ruby object. I will add a GitHub issue for that, and then we can implement it together next week. 😊